app/serializers/user_topic_bookmark_serializer.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class UserTopicBookmarkSerializer < UserPostTopicBookmarkBaseSerializer
+  attr_reader :topic
+
+  def initialize(obj, topic, opts)
+    super(obj, opts)
+    @topic = topic
+  end
+
+  # it does not matter what the linked post number is for topic bookmarks,
+  # on the client we always take the user to the last unread post in the
+  # topic when the bookmark URL is clicked
+  def linked_post_number
+    1
+  end
+
+  def deleted
+    topic.deleted_at.present? || topic.first_post.deleted_at.present?
+  end
+
+  def hidden
+    topic.first_post.hidden
```

Comment:
I think we need to preload the `first_post` too? Currently only `topic_users` are preloaded.